am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java

```diff
@@ -31,23 +36,40 @@ void setupTest() {
     }
 
     @Test
-    void filterResource_whenRowExistWithAccessorIdAndResourceId_ReturnPassedJsonObject() {
-        ams.createResourceAccess(createRecord(resourceId, ACCESSOR_ID, EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS));
+    void whenAddTwoRecordsWithSameResourceIdAndUserIdButDifferentAttribute_ReturnsTwoAttributes() {
+        ams.createResourceAccess(
+            createRecord(resourceId, ACCESSOR_ID, EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS, "/test"));
+        ams.createResourceAccess(
+            createRecord(resourceId, ACCESSOR_ID, EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS, "/name"));
 
         FilterResourceResponse result = ams.filterResource(ACCESSOR_ID, resourceId, DATA);
 
+        assertThat(result).toString().contains("test");
+        assertThat(result).toString().contains("name");
+    }
+
+    @Test
+    void whenRowExistWithAccessorIdAndResourceId_ReturnPassedFilteredJsonObject() {
```

Comment:
Wow. Having one test that covers filtering is way too little. It does me 0 (zero) confidence. Please largely expand on unit tests. Cover scenarios from confluence page and more.